### PR TITLE
Added simplified spoken forms for window snap

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
The existing snap spoken forms are rather verbose, requiring the user to think in terms of fractions.

In day-to-day use, I find it helpful to have "small" and "large" as shorter/simpler aliases. Let's add these as alternates; people can continue to use the old forms if they prefer them.

`"left third"` => `"left small"`
`"bottom right two thirds"` => `"bottom right large"`

I also added an action that takes the position name as a string, instead of the dataclass. This allows users to snap to a specific window position from a Talon file, which wasn't previously possible since you couldn't pass the `RelativeScreenPos` object from Talonscript (this was requested from a few users before).

`user.snap_window_to_position(position_name: str)`

### CODE REVIEW

File: core/windows_and_tabs/window_snap.py [hunk 1]
```diff
@@ -226,12 +226,13 @@ def __init__(self, left, top, right, bottom):
...
     "right third": RelativeScreenPos(2 / 3, 0, 1, 1),
     "left two thirds": RelativeScreenPos(0, 0, 2 / 3, 1),
-    "right two thirds": RelativeScreenPos(
-        1 / 3,
-        0,
-        1,
-        1,
-    ),
+    "right two thirds": RelativeScreenPos(1 / 3, 0, 1, 1),
+    # Alternate spoken forms for thirds
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:230]:
```suggestion
    # Alternate (simpler) spoken forms for thirds
```

File: core/windows_and_tabs/window_snap.py [hunk 2]
```diff
@@ -254,6 +255,17 @@ def __init__(self, left, top, right, bottom):
     "bottom left two thirds": RelativeScreenPos(0, 0.5, 2 / 3, 1),
     "bottom right two thirds": RelativeScreenPos(1 / 3, 0.5, 1, 1),
     "bottom center third": RelativeScreenPos(1 / 3, 0.5, 2 / 3, 1),
+    # Alternate spoken forms for sixths
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:258]:
```suggestion
    # Alternate (simpler) spoken forms for sixths
```

File: core/windows_and_tabs/window_snap.py [hunk 3]
```diff
@@ -275,11 +287,13 @@ class Actions:
     def snap_window(pos: RelativeScreenPos) -> None:
         """Move the active window to a specific position on-screen.
 
-        See `RelativeScreenPos` for the structure of this position.
-
-        """
+        See `RelativeScreenPos` for the structure of this position."""
         _snap_window_helper(ui.active_window(), pos)
 
+    def snap_window_to_position(pos_name: str) -> None:
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:293]:
```suggestion
    def snap_window_to_position(position_name: str) -> None:
```

File: core/windows_and_tabs/window_snap.py [hunk 4]
```diff
@@ -275,11 +287,13 @@ class Actions:
...
+        """Move the active window to a named position on-screen."""
+        actions.user.snap_window(_snap_positions[pos_name])
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:295]:
```suggestion
        actions.user.snap_window(_snap_positions[position_name])
```

File: core/windows_and_tabs/window_snap.py [hunk 5]
```diff
@@ -275,11 +287,13 @@ class Actions:
     def snap_window(pos: RelativeScreenPos) -> None:
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:287]: I know this isn't your code, but this is a tiny bit more pythonic

```suggestion
    def snap_window(position: RelativeScreenPos) -> None:
```

File: core/windows_and_tabs/window_snap.py [hunk 6]
```diff
@@ -275,11 +287,13 @@ class Actions:
...
         """Move the active window to a specific position on-screen.
 
-        See `RelativeScreenPos` for the structure of this position.
-
-        """
+        See `RelativeScreenPos` for the structure of this position."""
         _snap_window_helper(ui.active_window(), pos)
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:291]:
```suggestion
        _snap_window_helper(ui.active_window(), position)
```

File: core/windows_and_tabs/window_snap.py [hunk 7]
```diff
@@ -275,11 +287,13 @@ class Actions:
     def snap_window(pos: RelativeScreenPos) -> None:
         """Move the active window to a specific position on-screen.
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:288]:
```suggestion
        """Move the active window to a specific position given a `RelativeScreenPos` object.
```

File: core/windows_and_tabs/window_snap.py [hunk 8]
```diff
@@ -275,11 +287,13 @@ class Actions:
...
 
-        See `RelativeScreenPos` for the structure of this position.
-
-        """
+        See `RelativeScreenPos` for the structure of this position."""
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:290]: I would just delete this line entirely, the type information makes this redundant

File: core/windows_and_tabs/window_snap.py [hunk 9]
```diff
@@ -275,11 +287,13 @@ class Actions:
...
         _snap_window_helper(ui.active_window(), pos)
 
+    def snap_window_to_position(pos_name: str) -> None:
+        """Move the active window to a named position on-screen."""
```
Comment (phillco) [inline core/windows_and_tabs/window_snap.py:294]:
```suggestion
        """Move the active window to a specifically named position from `_snap_positions`."""
```

**Review by phillco — APPROVED** [review]
lgtm, mostly some description nits from the original function

**Comment by phillco** [comment]
I also punched up the description a little bit to try to explain the rationale for the new spoken forms

**Comment by phillco** [comment]
cc @jmegner  -- this will solve your problem of not being able to call the existing window snap code from `.talon`.

**Comment by AndreasArvidsson** [comment]
@phillco updated with your suggestions

**Review by phillco — APPROVED** [review]

**Review by fidgetingbits — APPROVED** [review]

**Comment by fidgetingbits** [comment]
Tried it on a gnome setup and works nicely, much more intuitive.

**Pull request merged**